### PR TITLE
test: fix TestIntakeAsync/semaphore_size_default flaky test

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

The test is using the default semaphore size (10) and sending a request with a lot of events followed by 9 concurrent requests. The first request (heavy.ndjson) is acquiring and releasing the semaphore 193 times and is also processed asynchronously on the server. Because of the async nature of the requests, the heavy request can hold more than a semaphore at the same time leading to a failure (queue is full) while processing the 9 concurrent requests.
The race condition can be avoided by processing the heavy request synchronously followed by 10 concurrent request to make sure the async processing on the intake handler is working.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes https://github.com/elastic/apm-server/issues/9173

### CODE REVIEW

File: systemtest/intake_async_test.go [hunk 1]
```diff
@@ -56,17 +56,17 @@ func TestIntakeAsync(t *testing.T) {
 		systemtest.CleanupElasticsearch(t)
 		srv := apmservertest.NewServerTB(t)
 
-		systemtest.SendBackendEventsAsyncPayload(t, srv.URL, `../testdata/intake-v2/errors.ndjson`)
+		systemtest.SendBackendEventsPayload(t, srv.URL, `../testdata/intake-v2/errors.ndjson`)
```
Comment (marclop) [inline systemtest/intake_async_test.go:59]: Let's revert this change, there's no need to process these synchronously since the line below asserts that the documents have been indexed to Elasticsearch, at which point, all the requests have been handled.
```suggestion
		systemtest.SendBackendEventsAsyncPayload(t, srv.URL, `../testdata/intake-v2/errors.ndjson`)
```

**Review by marclop — COMMENTED** [review]
Mostly LGTM. Just one small comment

File: systemtest/intake_async_test.go [hunk 2]
```diff
@@ -56,17 +56,17 @@ func TestIntakeAsync(t *testing.T) {
...
 		// Ensure the 5 errors are ingested.
 		systemtest.Elasticsearch.ExpectMinDocs(t, 5, "logs-apm.error-*", nil)
 
 		// Send a request with a lot of events (~1920) and expect it to be processed
 		// without any errors.
-		systemtest.SendBackendEventsAsyncPayload(t, srv.URL, `../testdata/intake-v2/heavy.ndjson`)
+		systemtest.SendBackendEventsPayload(t, srv.URL, `../testdata/intake-v2/heavy.ndjson`)
```
Comment (simitt) [inline systemtest/intake_async_test.go:65]: When this is the root cause, shouldn't https://github.com/elastic/apm-server/pull/9268/files#diff-74763070977bc9b95333184e98c4a318a3dcf2f6f6a62ba88de993bc65c726c1R41 also need to be changed then; to avoid that the `503` is issued due to the async requests using up the available resources?

Comment (marclop) [reply systemtest/intake_async_test.go:65]: `systemtest.SendBackendEventsPayload` will perform the Intake HTTP request synchronously, while `systemtest.SendBackendEventsAsyncPayload` will do that asynchronously.

The Queue is full errors are only thrown for async requests that can't be processed

File: systemtest/intake_async_test.go (hunk 1 above)
Comment (kruskall) [reply systemtest/intake_async_test.go]: Good catch!

**Review by marclop — APPROVED** [review]
LGTM! Thanks for fixing the flakiness.
